src/ecReduction.ml

```diff
@@ -120,15 +120,18 @@ module EqTest_base = struct
       | _, _ -> raise E.NotConv in
 
     let rec aux alpha e1 e2 =
-      e_equal e1 e2 || aux_r alpha e1 e2
+      (* If alpha is not empty then the test e_equal can be wrong *)
+         (e_equal e1 e2 && Mid.is_empty alpha)
+      || aux_r alpha e1 e2
 
     and aux_r alpha e1 e2 =
       match e1.e_node, e2.e_node with
       | Eint i1, Eint i2 ->
           BI.equal i1 i2
 
       | Elocal id1, Elocal id2 ->
-          EcIdent.id_equal (find alpha id1) id2
+          let t = EcIdent.id_equal (find alpha id1) id2 in
+          t
 
```

Comment:
What the point of that?

Reply:
I will remove it.

Reply:
I just inlined the title of #765. I don't want to have to click on the title to create the ChangeLog.